src/core/aws-net-websocket-registry.adb

```diff
@@ -494,14 +524,18 @@ package body AWS.Net.WebSocket.Registry is
          return Result : FD_Set.Socket_Set_Type do
             --  Add the signaling socket
 
-            FD_Set.Add (Result, Sig1, null, FD_Set.Input);
+            FD_Set.Add (Result, Sig1, No_UID, FD_Set.Input);
 
             --  Add watched sockets
 
             for Id of Watched loop
-               FD_Set.Add
-                 (Result,
-                  Registered (Id).all, Registered (Id), FD_Set.Input);
+               declare
+                  W : constant Object_Class := Registered (Id);
+               begin
+                  if not W.To_Free then
+                     FD_Set.Add (Result, W.all, W.Id, FD_Set.Input);
+                  end if;
```

Comment:
When I wrote a server (not based on AWS), I was also ensuring that sockets for which we got a notification in `epoll()` were removed from the `FD_Set` until we had actually read or written everything to/from it. Otherwise, that socket keeps popping up in `epoll` which is inefficient. Is something like this necessary here (perhaps a new Issue, though)

Reply:
Yes, for another issue. Let's keep this for the stability of the WS support.